src/Uno.UI/UI/Xaml/Controls/TextBox/OverlayTextBoxView/OverlayTextBoxViewExtension.skia.cs

```diff
@@ -165,6 +167,127 @@ public void UpdatePosition()
 		}
 	}
 
+	public void UpdateClip()
+	{
+		var clip = CalculateClipping(_contentElement.FindFirstChild<TextBlock>(), _contentElement.GetParent() as FrameworkElement);
+		_textBoxView?.SetClip((int)clip.clippingLeft, (int)clip.clippingRight, (int)clip.clippingTop, (int)clip.clippingBottom);
+
+		if (_lastSize.Width < clip.clippingLeft + clip.clippingRight || _lastSize.Height < clip.clippingTop + clip.clippingBottom)
+		{
+			_textBoxView?.SetVisibility(false);
+		}
+		else
+		{
+			_textBoxView?.SetVisibility(true);
+			if (_owner.TextBox?.FocusState != FocusState.Unfocused)
+			{
+				_textBoxView?.SetFocus();
+			}
+		}
+	}
+
+	private (double clippingLeft, double clippingRight, double clippingTop, double clippingBottom) CalculateClipping(FrameworkElement element, FrameworkElement? parent)
+	{
+		if (parent == null)
+		{
+			return (0, 0, 0, 0);
+		}
+
+		var elementRect = element.TransformToVisual(null).TransformBounds(new Rect(0, 0, element.ActualWidth, element.ActualHeight));
+		var parentRect = parent.TransformToVisual(null).TransformBounds(new Rect(0, 0, parent.ActualWidth, parent.ActualHeight));
+
+		if (parent is Windows.UI.Xaml.Controls.ScrollContentPresenter)
+		{
+			// SCP can lie wildly about bounds.
+			parentRect = elementRect;
+		}
```

Comment:
I'm really not liking this, but only ScrollContentPresenter seems to give weird Rects. I couldn't find something more robust for these calculations.